Modify TrajOptPlanner to allow a user to add custom terms to the trajectory optimization

carlosquinterop opened the pull request:
- It is now possible to inherit from the TrajOptPlanner class and override a plan() method. By doing this, the user can add custom terms (costs or constraints) into the trajectory optimization problem.
- The joint velocity cost is now incorporated in the new method addVelocityCost(). It used to be added in ProblemConstructionInfo() but since now the user can create custom problems, it is more convenient that everything is separated.
- A new script that shows how a user can create a custom trajopt planner for a UR5 robot and use it to add cartesian constraints for the end-effector for motion planning. The result is shown in the figure below:

![custom_trajopt_planner](https://user-images.githubusercontent.com/12850445/127406296-da01bbfc-d286-4318-b0da-f588ad227e9e.png)

zkingston commented on robowflex_tesseract/src/trajopt_planner.cpp:
```diff
@@ -406,6 +418,14 @@ TrajOptPlanner::PlannerResult TrajOptPlanner::plan(const SceneConstPtr &scene, c
+TrajOptPlanner::PlannerResult TrajOptPlanner::plan(const SceneConstPtr &scene,
+                                                   const robot_state::RobotStatePtr &start_state)
+{
+    RBX_ERROR("You need to implement virtual method TrajOptPlanner::plan() in your derived class");
```
probably better to throw an exception here rather than just a print

zkingston commented on robowflex_tesseract/scripts/ur5_custom_planning.cpp:
```diff
@@ -0,0 +1,156 @@
+ * This customized trajopt planner has end-effector pose constraints at specific timesteps of the trajectory.
+ */
+
+namespace robowflex
```
the custom class doesn't need to be in the robowflex namespace, you can just have it in the global

zkingston reviewed (changes requested):
Some small nits. We should also add a trajopt docker to the CI to check compilation + maybe a small test script.

carlosquinterop commented:
> Some small nits. We should also add a trajopt docker to the CI to check compilation + maybe a small test script.

I've addressed the specific comments and also updated the README file accordingly. Regarding the docker to the CI, I'll go ahead and open an issue with this so we can address it in a different PR
